src/scenes/NftDetailPage/NavigationHandle.tsx

```diff
@@ -20,23 +21,26 @@ const HOVER_TEXT = new Map<number, string>([
 
 type Props = {
   direction: Directions;
+  username: string;
+  collectionId: string;
   nftId: string;
 };
 
-function NavigationHandle({ direction, nftId }: Props) {
+function NavigationHandle({ direction, username, collectionId, nftId }: Props) {
   const arrow = useMemo(() => ARROWS.get(direction) ?? '', [direction]);
 
   const hoverText = useMemo(() => HOVER_TEXT.get(direction) ?? '', [direction]);
 
-  const { push } = useRouter();
+  const { replace } = useRouter();
   const handleOnClick = useCallback(() => {
     // TODO(Terence): Figure out how to get this state across since Next doesn't support navigation state
     // void push(nftId, { state: { collection: [] } });
-    void push({ pathname: nftId });
-  }, [nftId, push]);
+    // void push({ pathname: nftId });
+    void replace(`/${username}/${collectionId}/${nftId}`);
```

Comment:
Is replacing history acceptable here, or is `push` needed? 

If so, we will need to be creative in how to handle `useBackButton` via clicking 'Back to Gallery' which by default will just go back in the user's history.

Reply:
oooo good question, i think we should use `replace` for that reason